source/lib/ze_lib.h

```diff
@@ -34,22 +36,130 @@ namespace ze_lib
         context_t();
         ~context_t();
 
+    //////////////////////////////////////////////////////////////////////////
+    std::string to_string(const ze_result_t result) {
```

Comment:
This work, but have you considered using stringify w/preprocessor macros? 
Which may also work to not miss future ZE_RESULT additions

Trying to avoid a future maintenance issue